File: react/car_rental_react/src/components/Vehicle.js
import { useState } from "react";
import VehicleService from "../services/VehicleService";

const Vehicle = () => {
    const [cars, setCars] = useState([]);
    const vehicleService = VehicleService();

    const show = async () => {
        try {
            const response = await vehicleService.adminDashBoard();
            setCars(response.data); // ✅ use response.data if Axios
        } catch (error) {
            console.error("Error fetching cars", error);
        }
    };

    return (
        <div>
            <table border="3" align="center">
                <thead>
                    <tr>
                        <th>Vehicle Id</th>
                        <th>Make</th>
                        <th>Model</th>
                        <th>Year</th>
                        <th>Daily Rate</th>
                        <th>Status</th>
                        <th>Passenger Capacity</th>
                        <th>Engine Capacity</th>
                    </tr>
                </thead>
                <tbody>
                    {cars.map((item) => (
                        <tr key={item.vehicleID}>
                            <td>{item.vehicleID}</td>
                            <td>{item.make}</td>
                            <td>{item.model}</td>
                            <td>{item.year}</td>
                            <td>{item.dailyRate}</td>
                            <td>{item.status}</td>
                            <td>{item.passengerCapacity}</td>
                            <td>{item.engineCapacity}</td>

                        </tr>
                    ))}
                </tbody>
            </table>
            <input type="button" value="Show Vehicle" onClick={show} /> <br /><br />
        </div>
    );
};

export default Vehicle;

File: Asp.NetCore/Aug_25_Car_Rental_Core/Aug_25_Car_Rental_Core/Models/Vehicle.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Aug_25_Car_Rental_Core.Models
{
    public class Vehicle
    {
        [Key]
        [Column("vehicleId")]
        public int VehicleID { get; set; }

        [Column("make")]
        public string? Make {  get; set; }

        [Column("model")]
        public string? Model { get; set; }

        [Column("year")]
        public string? Year { get; set; }

        [Column("dailyRate")]
        public string? DailyRate { get; set; }

        [Column("status")]
        public string? Status { get; set; }

        [Column("passengerCapacity")]
        public string? PassengerCapacity { get; set; }

        [Column("engineCapacity")]
        public string? EngineCapacity { get; set; }

    }
}
